src/types.rs

```diff
@@ -186,6 +186,56 @@ impl<T, E> ExpectUtf8<T> for core::result::Result<T, E> {
     }
 }
 
+#[cfg(feature = "engine")]
+pub trait SdkExpect<T> {
+    fn sdk_expect(self, msg: &str) -> T;
+}
+
+#[cfg(feature = "engine")]
+impl<T> SdkExpect<T> for Option<T> {
+    fn sdk_expect(self, msg: &str) -> T {
+        match self {
+            Some(t) => t,
+            None => sdk::panic_utf8(msg.as_ref()),
+        }
+    }
+}
+
+#[cfg(feature = "engine")]
+impl<T, E> SdkExpect<T> for core::result::Result<T, E> {
+    fn sdk_expect(self, msg: &str) -> T {
+        match self {
+            Ok(t) => t,
+            Err(_) => sdk::panic_utf8(msg.as_ref()),
+        }
+    }
+}
+
+#[cfg(feature = "engine")]
+pub trait SdkUnwrap<T> {
+    fn sdk_unwrap(self) -> T;
+}
+
+#[cfg(feature = "engine")]
+impl<T> SdkUnwrap<T> for Option<T> {
+    fn sdk_unwrap(self) -> T {
+        match self {
+            Some(t) => t,
+            None => sdk::panic_utf8("ERR_UNWRAP".as_bytes()),
+        }
+    }
+}
+
+#[cfg(feature = "engine")]
+impl<T, E: AsRef<[u8]>> SdkUnwrap<T> for core::result::Result<T, E> {
+    fn sdk_unwrap(self) -> T {
+        match self {
+            Ok(t) => t,
+            Err(e) => sdk::panic_utf8(e.as_ref()),
+        }
+    }
+}
+
```

Comment:
To be fair, this should be somewhere else. I wouldn't exactly call them types.

Reply:
Perhaps, yes, I agree!. Please also note we have similar traits in the same `types` file.
So if you don't mind, I suggest merging this PR and leave this part for the next one with some refactoring and structuring changes.

Reply:
I agree with @sept-en that we should not block this PR over this sort of code organization. If it really is `P0` as the tag on the PR says then getting through and fixing minor comments later seems better than bikeshedding on it now.

Reply:
@joshuajbouw @birchmd created issue #134 for that. Please, let's merge this PR ASAP if possible :)